File: dotnet/Middleware.Contracts/Events/BillingAssociationCreatedEvent.cs
using NServiceBus;

namespace Middleware.Contracts.Events;

public sealed class BillingAssociationCreatedEvent : IEvent
{
    public string IssuanceId { get; set; } = string.Empty;
    public string AccountId { get; set; } = string.Empty;
    public string AccountServiceRequestNumber { get; set; } = string.Empty;
    public string BillingChannel { get; set; } = "DirectBill";
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
}

File: java/common/src/main/java/com/ais/middleware/common/events/billing/BillingAssociationCreatedEvent.java
package com.ais.middleware.common.events.billing;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;

/**
 * Published by BillingAndFinanceManagement after creating the BillingAccount record and associating it with the external billing system.
 * This is one of the two events required for the IssuanceSaga join condition (BR-PIL-003).
 */
public record BillingAssociationCreatedEvent(
        String accountId,
        String issuanceId,
        String accountServiceRequestNumber,
        BillingChannel billingChannel,
        OffsetDateTime createdAt
) {
    public enum BillingChannel {
        DirectBill, AgencyBill, Digital
    }

    @JsonCreator
    public BillingAssociationCreatedEvent(
            @JsonProperty("accountId") String accountId,
            @JsonProperty("issuanceId") String issuanceId,
            @JsonProperty("accountServiceRequestNumber") String accountServiceRequestNumber,
            @JsonProperty("billingChannel") BillingChannel billingChannel,
            @JsonProperty("createdAt") OffsetDateTime createdAt
    ) {
        this.accountId = accountId;
        this.issuanceId = issuanceId;
        this.accountServiceRequestNumber = accountServiceRequestNumber;
        this.billingChannel = billingChannel;
        this.createdAt = createdAt;
    }
}
